node/src/accountant/db_access_objects/sent_payable_dao.rs

```diff
@@ -574,166 +552,157 @@ mod tests {
     }
 
     #[test]
-    fn can_retrieve_txs_to_retry() {
+    fn tx_can_be_retrieved_by_hash() {
         let home_dir =
-            ensure_node_home_directory_exists("sent_payable_dao", "can_retrieve_txs_to_retry");
+            ensure_node_home_directory_exists("sent_payable_dao", "tx_can_be_retrieved_by_hash");
         let wrapped_conn = DbInitializerReal::default()
             .initialize(&home_dir, DbInitializationConfig::test_default())
             .unwrap();
         let subject = SentPayableDaoReal::new(wrapped_conn);
-        let old_timestamp = current_unix_timestamp() - 60; // 1 minute old
-        let tx1 = TxBuilder::default()
-            .hash(H256::from_low_u64_le(3))
-            .timestamp(old_timestamp)
-            .status(TxStatus::Pending)
-            .build();
-        let tx2 = TxBuilder::default()
-            .hash(H256::from_low_u64_le(4))
-            .timestamp(old_timestamp)
-            .status(TxStatus::Succeeded(TransactionBlock {
-                block_hash: Default::default(),
-                block_number: Default::default(),
-            }))
-            .build();
-        // TODO: GH-631: Instead of fetching it from SentPayables, fetch it from the FailedPayables table
-        let tx3 = TxBuilder::default() // this should be picked for retry
-            .hash(H256::from_low_u64_le(5))
-            .timestamp(old_timestamp)
-            .status(TxStatus::Failed)
-            .build();
-        let tx4 = TxBuilder::default() // this should be picked for retry
-            .hash(H256::from_low_u64_le(6))
-            .status(TxStatus::Failed)
-            .build();
-        let tx5 = TxBuilder::default()
-            .hash(H256::from_low_u64_le(7))
-            .timestamp(old_timestamp)
-            .status(TxStatus::Pending)
-            .build();
+        let tx1 = TxBuilder::default().hash(make_tx_hash(1)).build();
+        let tx2 = TxBuilder::default().hash(make_tx_hash(2)).build();
         subject
-            .insert_new_records(&vec![tx1, tx2, tx3.clone(), tx4.clone(), tx5])
+            .insert_new_records(&vec![tx1.clone(), tx2.clone()])
             .unwrap();
 
-        let result = subject.retrieve_txs(Some(RetrieveCondition::ToRetry));
+        let result = subject.retrieve_txs(Some(ByHash(vec![tx1.hash])));
 
-        assert_eq!(result, vec![tx3, tx4]);
+        assert_eq!(result, vec![tx1]);
```

Comment:
This function potentially can return a collection of values and so it would be nice to see it in the assertion. Maybe two transactions retrieved and three in the database in total.